app.py

```diff
@@ -52,6 +52,15 @@ def experience():
 
     return jsonify({})
 
+@app.route('/resume/experience/<int:index>', methods=['GET'])
+def experience_by_index(index):
+    '''
+    Handle experience requests by index
+    '''
+    if 0 <= len(data['experience']) and index < len(data['experience']):
+        return jsonify(data['experience'][index])
+    return jsonify({'error': 'Index out of range'}), 404
```

Comment:
nit: Let's say experience not found to make it a bit more generic and not give away the implementation detail of checking by index

Reply:
I made the change to the error message as requested.